### ISSUE TITLE
[documentation] Function of "Don't know" option

### ISSUE THREAD

**Issue by Gitoffthelawn** [M66B/XPrivacy#2306]
Hi Marcel.  I noticed that a piece of significant information, namely the function of the `Don't know` button (between `Allow` and `Deny`), is missing from the README.

I started writing an explanation of it, which caused me to realize that I do not fully understand the function of `Don't know`.

Can you provide details about it?  If so, I'll write something up and add it to the README.

Exactly what does selecting `Don't know` do, and how does it differ from `Allow` and `Deny`?

**Comment by M66B** [comment]
Don't know will deny non dangerous restrictions and allow dangerous restrictions for 15 seconds.

**Comment by Gitoffthelawn** [comment]
It seems to do more than that (or perhaps there is a bug).  When I click it when apps try to access the `contacts/data` function, it behaves differently than clicking either `Allow` or `Deny`.

Clicking `Allow` always allows access to all contacts (the whitelist is ignored).  Clicking `Deny` always denies access to all contacts.  Clicking `Don't know` only allows access to contacts in the whitelist.

**Comment by M66B** [comment]
It does what I have described.

**Comment by Gitoffthelawn** [comment]
Then I think there is a bug.  Because what you describe is not what XPrivacy actually does!

**Comment by M66B** [comment]
I think a bug of unlikely after so much time and so much users, but you'll never know.

Enabling debug logging, which will allow down your device a bit, and inspecting the logcat might tell.

**Comment by Gitoffthelawn** [comment]
Was doing it as you were typing. :-)

Any ideas why this is what I witness?: For the `contacts/data` function, clicking `Allow` always allows access to all contacts (the whitelist is ignored). Clicking `Deny` always denies access to all contacts. Clicking `Don't know` only allows access to contacts in the whitelist.

**Comment by M66B** [comment]
https://github.com/M66B/XPrivacy/blob/master/src/biz/bokhorst/xprivacy/PrivacyService.java#L1968

**Comment by Gitoffthelawn** [comment]
Since `contacts/data` is not a dangerous function, `Don't know` should deny it.  But it is allowing all contacts in the whitelist.  To the contrary, `Deny` does not allow contacts in the whitelist.

**Comment by M66B** [comment]
Contacts in the whitelist will always be allowed, regardless which of the three buttons you use.

**Comment by Gitoffthelawn** [comment]
Marcel, I owe you an apology.  I'm sorry.  I made an error.  I'm not sure how it happened.  Clicking `Deny` *does* allow the whitelist items.  I checked several times and I thought it did not.  Either I was dealing with a caching issue, or I just flat out made a mistake.  I'm so very sorry I wasted your time on that.

I'll write up one or two new items for the README and submit a pull request.  Again, I'm very sorry for my error.
